frontends/web/src/components/dialog/dialog.tsx

```diff
@@ -175,26 +182,28 @@ class Dialog extends Component<Props, State> {
         const isSlim = slim ? style.slim : '';
         const isCentered = centered && !onClose ? style.centered : '';
         return (
-            <div className={[style.overlay].join(' ')} ref={this.setOverlay}>
-                <div className={[style.modal, isSmall, isMedium, isLarge].join(' ')} ref={this.setModal}>
+            <div className={style.overlay} ref={element => this.overlay = element}>
+                <div
+                    className={[style.modal, isSmall, isMedium, isLarge].join(' ')}
+                    ref={element => this.modal = element}>
                     {
                         title && (
                             <div className={[style.header, isCentered].join(' ')}>
                                 <h3 class={style.title}>{title}</h3>
-                                {
-                                    onClose && (
-                                        <button className={style.closeButton} onClick={this.deactivate} disabled={disabledClose}>
-                                            <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round">
-                                                <line x1="18" y1="6" x2="6" y2="18"></line>
-                                                <line x1="6" y1="6" x2="18" y2="18"></line>
-                                            </svg>
-                                        </button>
-                                    )
-                                }
+                                { onClose ? (
+                                    <button className={style.closeButton} onClick={this.deactivate} disabled={disabledClose}>
+                                        <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round">
+                                            <line x1="18" y1="6" x2="6" y2="18"></line>
+                                            <line x1="6" y1="6" x2="18" y2="18"></line>
+                                        </svg>
+                                    </button>
+                                ) : null }
                             </div>
                         )
                     }
-                    <div className={[style.contentContainer, isSlim].join(' ')} ref={this.setModalContent}>
+                    <div
+                        className={[style.contentContainer, isSlim].join(' ')}
+                        ref={element => this.modalContent = element}>
```

Comment:
Is this like known pattern to prefer inlined code in react rendering instead of functions? To me, `ref={this.setModalContent}` is subjectively better.

Or is there some kind of technical reason to do inline `ref={element => ... }`?

Reply:
Same, I preferred having the function not defined in the render function.

> **Caveats with callback refs**
If the ref callback is defined as an inline function, it will get called twice during updates, first with null and then again with the DOM element. This is because a new instance of the function is created with each render, so React needs to clear the old ref and set up the new one. You can avoid this by defining the ref callback as a bound method on the class, but note that it shouldn’t matter in most cases.

https://reactjs.org/docs/refs-and-the-dom.html#caveats-with-callback-refs

In the long term this is this will change anyway when we use hooks. For now I find is more readable, but can change back why do you find it subjectively better?

Reply:
It's ok. So, there's a technical explanation for this. Just wanted to know. :+1:

Reply:
It is in theory slightly less performant as `a new instance of the function is created with each render`, unless there is optimizing step in the build process, I am not aware of any but there could ofc be something.

In the future we can use useRef hook
https://reactjs.org/docs/hooks-reference.html#useref